security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -1056,6 +1170,59 @@ func (sc *SecretCache) sendRetriableRequest(ctx context.Context, csrPEM []byte,
 	return []string{exchangedToken}, nil
 }
 
+func (sc *SecretCache) sendRetriableCSRRequestWithoutToken(ctx context.Context, csrPEM []byte, connKey ConnKey, isCSR bool) ([]string, error) {
```

Comment:
Again this seems a huge duplicate of the sendRetriableCSRRequest() below, please refactor to share the common part instead of copying everything again.